src/libs/Network/NetworkEvents.js

```diff
@@ -0,0 +1,34 @@
+import CONST from '../../CONST';
+import createCallback from '../createCallback';
+
+const [getLogger, registerLogHandler] = createCallback();
+const [triggerConnectivityResumed, onConnectivityResumed] = createCallback();
+const [onRequest, registerRequestHandler] = createCallback();
```

Comment:
I find this naming a bit confusing. We have two separate naming conventions both being used:

1. `onXYZ` to emit the event and `registerXYZHandler` to register a handler for that event.
1. `triggerXYZ` to emit the event and `onXYZ` to register a handler for that event.

We should choose one and be consistent. I would recommend `2`, or alternatively could do `triggerXYZ` and `registerXYZHandler`

Reply:
I think we should do the above ^ in any event, but you _could_ also take this a step further and create a `NETWORK_EVENTS` constant like:

```js
NETWORK_EVENTS = {
    REQUEST: 'request',
    RESPONSE: 'response',
    ERROR: 'error',
    CONNECTIVITY_RESUMED: 'connectivityResumed',
    CONNECTIVITY_CHECK: 'connectivityCheck',
};
```

And then refactor this lib like so:

```js
const events = _.reduce(
    CONST.NETWORK_EVENTS,
    (memo, eventName) => ({
        ...memo,
        [eventName]: createCallback(),
    }),
    {},
);

function emit(name, payload) {
    const event = events[name];
    if (!event) {
        throw new Error(`The ${name} network event does not exist!`);
    }
    
    const eventTrigger = event[0];
    eventTrigger(payload);
}

function registerHandler(name, callback) {
    const event = events[name]
    if (!event) {
        throw new Error(`The ${name} network event does not exist!`);
    }
    
    const registerEventHandler = event[1];
    registerEventHandler(name, callback);
}

export {
    emit,
    registerHandler,
};
```

And then use the refactored lib like this:

```js
NetworkEvents.registerHandler(CONST.NETWORK_EVENTS.ERROR, (error) => Log.alert(error.message));
NetworkEvents.emit(CONST.NETWORK_EVENTS.ERROR, {message: 'Some error message'});
```

Reply:
Solid idea. Can see how it cleans stuff up.

thought: It looks like we'd be able to register n handlers for a given event and that might lead one to make a mistake that they could add extra handlers

proposal: Let's clean up the names for now and do a proper `EventEmitter` class that can handle multiple callbacks in the next PR?

Reply:
> Let's clean up the names for now and do a proper EventEmitter class that can handle multiple callbacks in the next PR?

Sounds great! I'm happy to approve whatever level of cleanup you want to do 😄